images/operator-sdk/Dockerfile

```diff
@@ -5,4 +5,7 @@ RUN microdnf install -y golang make which
 ARG BIN=bin/operator-sdk
 COPY $BIN /usr/local/bin/operator-sdk
 
+RUN GOBIN=/usr/local/bin/ GO111MODULE=on go get sigs.k8s.io/kustomize/kustomize/v3@v3.5.4 \
+ && GOBIN=/usr/local/bin/ GO111MODULE=on go get sigs.k8s.io/controller-tools/cmd/controller-gen@v0.3.0
```

Comment:
@jmrodri - small doubt. Is there any specific reason for picking controller-gen 0.3.0? We could have the latest version instead (0.4.1)

Reply:
As long as it doesn't break support in the latest version, I agree let's go latest.

Reply:
@varshaprasad96  @aemperador updated to v0.4.1. The one thing I noticed is that the v2 plugin uses v0.3.0 but v3 uses v0.4.1. 

```
testdata/go/v2/memcached-operator/Makefile:106: go get sigs.k8s.io/controller-tools/cmd/controller-gen@v0.3.0 ;\
testdata/go/v3/memcached-operator/Makefile:104: $(call go-get-tool,$(CONTROLLER_GEN),sigs.k8s.io/controller-tools/cmd/controller-gen@v0.4.1)
```
Is this going to be a problem? Also, I noticed `kustomize` is different between `v2` and `v3`:

```
testdata/go/v2/memcached-operator/Makefile:121: go get sigs.k8s.io/kustomize/kustomize/v3@v3.5.4 ;\
testdata/go/v3/memcached-operator/Makefile:109: $(call go-get-tool,$(KUSTOMIZE),sigs.k8s.io/kustomize/kustomize/v3@v3.8.7)
```

Reply:
@estroz thoughts? ^^^

Reply:
Three ways to tackle this:
1. Make an image for each plugin version with the appropriate tool version.
	- `quay.io/operator-framework/operator-sdk-go-v2:v1.3.0` containing all go/v2 tools
2. Download multiple versions of tool binaries into the same image, with each binary name containing its version, ex. `kustomize-3-5-4`.
	- `quay.io/operator-framework/operator-sdk-go:v1.3.0` containing all tools for go/v2 and go/v3 with versioned binary names
3. Make an image for dependency tool and version supported.
	- `quay.io/operator-framework/kustomize:v3.5.4`, `quay.io/operator-framework/kustomize:v3.8.7`

I personally prefer (3) because it scales the best, produces the smallest images, and is plugin-agnostic. The major problem with the first two is that the kustomize version (and controller-gen's for Go operators) is dependent on plugin version (and type, even though a divergence hasn't happened yet), while a given operator-sdk version supports multiple plugins. We'd have to maintain a build matrix of `operator-sdk version x plugin type x plugin version`, for which tagging would not scale as well (see examples) and we'd have to rebuid these images each time a new operator-sdk version was released.

Reply:
Example Dockerfile for kustomize:
```dockerfile
FROM registry.access.redhat.com/ubi8/ubi-minimal:8.3-227

ARG ARCH
RUN curl -sSLo - https://github.com/kubernetes-sigs/kustomize/releases/download/kustomize/v3.5.4/kustomize_v3.5.4_linux_${ARCH}.tar.gz | \
tar xzf - -C /

ENTRYPOINT ["/kustomize"]
```
Build with:
```sh
docker build \
	-f images/kustomize/Dockerfile \
	-t quay.io/operator-framework/kustomize:v3.5.4 \
	--build-arg ARCH=$(go env GOARCH) \
	images/kustomize
```